client/locations-list-map.js

```diff
@@ -550,168 +809,155 @@ function onFilterChange(data, prefix, idx, selected, filters) {
-    const mapBounds = map.getBounds();
-
-    if (mapBounds && currentViewportCenter) {
-      const currentLat = mapBounds.getCenter().lat();
-      const currentLng = mapBounds.getCenter().lng();
+  const facilityTypes = [];
 
-      if (currentLat !== currentViewportCenter.lat || currentLng !== currentViewportCenter.lng) {
-        refreshList(data, filters);
-      }
+  for (const item of Object.keys(filters.orgTypes)) {
+    const orgType = filters.orgTypes[item];
+    facilityTypes.push({
+      value: orgType.value,
+      text: orgType.name,
+      selected: orgType.isSet,
+    });
+  }
 
-      currentViewportCenter = {
-        lat: currentLat,
-        lng: currentLng
-      }
-    }
-  });
+  if (facilityTypes.length > 0) {
+    const facilityTypeSelect = new Selectr('#facility-type-select', {
+      customClass: 'ftm-select',
+      data: facilityTypes,
+      multiple: true,
+      searchable: false,
+      placeholder: $.i18n('ftm-facility-type'),
+    });
 
-  const mapBounds = map.getBounds();
+    facilityTypeSelect.on('selectr.select', (option) => {
+      onFilterChange(data, 'orgTypes', option.idx, true, filters);
+      sendEvent('filters', 'orgTypes', option.value);
+    });
 
-  if (mapBounds) {
-    const mapCenter = mapBounds.getCenter();
-    currentViewportCenter = {
-      lat: mapCenter.lat(),
-      lng: mapCenter.lng()
-    };
+    facilityTypeSelect.on('selectr.deselect', (option) => {
+      onFilterChange(data, 'orgTypes', option.idx, false, filters);
+    });
+  } else {
+    $('#facility-type-select').parent().remove();
   }
+}
 
-  showMarkers(data, filters);
+function getDataJsonUrl(filename) {
+  return `/${filename}`;
+}
 
-  // Initialize autosuggest/search field above the map.
-  initMapSearch(data, filters);
+function getCountryDataFilename(country) {
+  // Always use country-specific data.json file
+  return `data-${country}.json`;
+}
 
-  if (isCountryPath()) {
-    loadOtherCountries();
+// Loads data file from url and assigns into object given by dataToStore
+function loadDataFile(url, dataToStore) {
+  $.getJSON(
+    url,
+    (result) => {
+      Object.assign(dataToStore, toDataByLocation(result));
+
+      // opacity value matches what's in css for the .secondarycluster class -
+      // can set a css class for the clusters, but not for individual pins.
+      gOtherMarkers.push(
+        ...getMarkers(dataToStore, {}, null, SECONDARY_MARKER_OPTIONS).outOfFilters,
+      );
+      updateClusters(null, gSecondaryCluster);
+    },
+  );
+}
+
+function loadOtherCountries() {
+  const countryCodes = Object.keys(countries);
+
+  for (const code of countryCodes) {
+    if (code !== currentCountry) {
+      countryData[code] = {};
+      loadDataFile(getDataJsonUrl(getCountryDataFilename(code)), countryData[code]);
+    }
   }
 }
 
-/**********************************
- * BEGIN MAP SEARCH FUNCTIONALITY *
- **********************************/
+let currentViewportCenter = {};
```

Comment:
Move up and add 'g' prefix with the other globals?

Reply:
done.